### PR TITLE
Functional test recipes for PMC and HostName modules

### PR DESCRIPTION
**Pull request by hachotze**
## Description
- Added functional test recipes for PMC and HostName modules.
- Removed non-required WaitSeconds parameters from Networking and TPM recipes.
- Added zero case to check for 'payload too large' in HostNameBase::Set().
## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/RecipeInvoker.cpp [hunk 1]
```diff
@@ -96,6 +96,8 @@ void RecipeInvoker::TestBody()
         std::cout << "Waiting for " << m_recipe.m_waitSeconds << " seconds" << std::endl;
         std::this_thread::sleep_for(std::chrono::seconds(m_recipe.m_waitSeconds));
     }
+
+    std::remove("/etc/osconfig/osconfig_commandrunner.cache");
```
Comment (robertschaedler3) [inline src/modules/test/RecipeInvoker.cpp:100]: Is it possible to save a copy of this file before running the tests, then restore the cache instead of removing it completely?

Comment (robertschaedler3) [reply src/modules/test/RecipeInvoker.cpp:100]: Small corner case to consider that could impact a dev environment significantly: 

1. Send Reboot/Shutdown via the IoTHub
2. Stop OSConfig (agent/platform) 
3. Run `modulestest`
4. Restart OSConfig **<-** *this will cause your dev machine to Reboot/Shutdown*

**Comment by robertschaedler3** [comment]
It looks like renaming the hostname test recipe is breaking the build:

```
-- Adding module 'Hostname' to the build.
CMake Error at modules/CMakeLists.txt:22 (message):
  Module Hostname must have a recipe file at:
  /__w/azure-osconfig/azure-osconfig/src/modules/test/recipes/HostnameTests.json
Call Stack (most recent call first):
  modules/CMakeLists.txt:29 (add_module_custom_target)
  modules/CMakeLists.txt:41 (add_module)
```

File: src/modules/test/RecipeInvoker.cpp (hunk 1 above)
Comment (AhmedBM) [reply src/modules/test/RecipeInvoker.cpp:100]: yes, it would affect it if that scenario was run. But this is a temporary workaround until we have a good setup/teardown solution

**Review by AhmedBM — COMMENTED** [review]
I think the recipe rename will break the recipe configuration generation. You might have tyo resort to using `add_module_custom_target` in order to be explicit for the naming of your targets/mim/recipes

File: src/modules/test/RecipeInvoker.cpp (hunk 1 above)
Comment (MariusNi) [reply src/modules/test/RecipeInvoker.cpp:100]: This is now done in main, please remove from this PR. Do not merge with this in, will duplicate the change (the one in main is on a different line)

File: src/modules/test/recipes/HostNameTests.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    "ObjectName": "name",
+    "Desired": true,
+    "Payload": "{}",
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "CommandRunner",
+    "ObjectName": "commandArguments",
+    "Desired": true,
+    "Payload": "{\"commandId\": \"saveHostNameInTempFile\", \"arguments\": \"echo $(hostname) > HostNameTestsTemp.txt\", \"timeout\": 0, \"singleLineTextResult\": true, \"action\": 3}",
```
Comment (MariusNi) [inline src/modules/test/recipes/HostNameTests.json:55]: >"Payload": "{\"commandId\": \"saveHostNameInTempFile\", \"arguments\": \"echo $(hostname) > HostNameTestsTemp.txt\", \"timeout\": 0, \"singleLineTextResult\": true, \"action\": 3}", [](http://example.com/codeflow?start=4&length=181)

There are no space separators in actual payloads. Suggest removing, here and elsewhere.

Comment (hachotze) [reply src/modules/test/recipes/HostNameTests.json]: Agree, addressed this in the latest iteration.

File: src/modules/test/RecipeInvoker.cpp (hunk 1 above)
Comment (hachotze) [reply src/modules/test/RecipeInvoker.cpp]: Removed this in the latest iteration.

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

File: src/modules/test/recipes/PmcTests.json [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    "ObjectName": "desiredState",
+    "Desired": true,
+    "Payload": "{}",
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "CommandRunner",
+    "ObjectName": "commandArguments",
+    "Desired": true,
+    "Payload": "{\"commandId\":\"addMicrosoftPublicKey\",\"arguments\":\"apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EB3E94ADBE1229CF\",\"timeout\":0,\"singleLineTextResult\":true,\"action\":3}",
```
Comment (AhmedBM) [inline src/modules/test/recipes/PmcTests.json:49]: why are we using commandRunner to add the key? PMC supports the `gpgKeys` see https://github.com/Azure/azure-osconfig/blob/main/src/modules/pmc/README.md

Comment (hachotze) [reply src/modules/test/recipes/PmcTests.json]: Agree, removed this check.

**Review by AhmedBM — APPROVED** [review]
Looks good 👌
